### PR TITLE
Use base58 encoding from ledger SDK

### LINKED ISSUE #46: Use SDK base58_encoder instead of our own
Ledger SDK provides base58_encoder which makes base58_encoding.h implementation in the app redundant. Use the SDK version.

### PR DESCRIPTION
**Pull request by ajinkyaraj-23**
fixes #46

### CODE REVIEW

**Review by spalmer25 — APPROVED** [review]
LGTM

File: src/main.c [hunk 1]
```diff
@@ -8,6 +8,9 @@
 #include "memory.h"
 
 __attribute__((noreturn)) void app_main(void) {
+    init_globals();
+    ui_initial_screen();
+
```
Comment (spalmer25) [inline src/main.c:13]: I've added the removed initialization code.

File: src/globals.h [hunk 2]
```diff
@@ -130,10 +129,6 @@ typedef struct {
 
 extern globals_t global;
 
-extern unsigned int app_stack_canary;  // From SDK
-
```
Comment (spalmer25) [inline src/globals.h:134]: I've removed the unused `app_stack_canary`

File: src/globals.c [hunk 3]
```diff
@@ -16,23 +16,12 @@
...
     memset(&global.apdu, 0, sizeof(global.apdu));
 }
 
 void init_globals(void) {
     memset(&global, 0, sizeof(global));
-
-    memset(&G_ux, 0, sizeof(G_ux));
-    memset(&G_ux_params, 0, sizeof(G_ux_params));
-
-    memset(G_io_seproxyhal_spi_buffer, 0, sizeof(G_io_seproxyhal_spi_buffer));
```
Comment (spalmer25) [inline src/globals.c:35]: I've removed the initialization done in the SDK

File: misra.json [hunk 4]
```diff
@@ -2,6 +2,6 @@
     "script": "misra.py",
     "args": [
         "--rule-texts=misra.md",
-        "--suppress-rules 2.7,3.1,7.2,7.4,8.2,8.4,8.9,8.14,9.2,10.1,10.3,10.4,10.5,10.6,11.1,11.3,11.5,11.8,12.1,12.3,13.3,13.4,14.2,14.4,15.5,16.3,16.4,16.5,16.6,17.7,17.8,18.4,18.8,19.2,20.5,20.7,21.15,21.16"
+        "--suppress-rules 2.7,3.1,7.2,7.4,8.2,8.4,8.9,8.14,9.2,10.1,10.3,10.4,10.5,10.6,11.1,11.3,11.5,11.8,12.1,13.3,13.4,14.2,14.4,15.5,16.3,16.4,16.5,16.6,17.7,17.8,18.4,18.8,19.2,20.5,20.7,21.15,21.16"
```
Comment (spalmer25) [inline misra.json:5]: I've removed the `misra-rule` 12.3 from the unchecked rules as it's checked now:
```
Rule 12.3 Advisory
The comma operator should not be used
```

**Review by spalmer25 — APPROVED** [review]

File: src/globals.h [hunk 5]
```diff
@@ -99,7 +99,6 @@ typedef struct {
         bool is_blank_screen;
     } dynamic_display;
 
-    void *stack_root;
```
Comment (spalmer25) [inline src/globals.h:102]: I've removed the unused `stack_root`

**Review by spalmer25 — APPROVED** [review]

**Comment by ajinkyaraj-23** [comment]
> I've written below what I've modified so that you can review your own PR.

Excellent. Now it works.

File: src/main.c (hunk 1 above)
Comment (ajinkyaraj-23) [reply src/main.c:13]: This was the missing part.

**Pull request merged**